### PR TITLE
feat(efb): in-flight refueling capabilities

### PR DESCRIPTION
**Pull request by Taz5150**
Fixes #[issue_no]

## Summary of Changes

As part of the whole fuel system overhaul, the EFB fuel page will become **master** for refueling/ defueling purposes over the in-game UI menu which will be overridden at some point. Through the EFB you could only refuel/ defuel in a C&D situation and not during other flight phases. Since the in-game UI will not be operational at some point, we have to give the opportunity to refuel/ defuel at any time.

This PR adds the possibility to refuel/ defuel at any given time ONLY with the Instant option, while Fast and Real will only be allowed on ground with engines stopped.

## Screenshots (if necessary)

Some visual changes accompany this feature:

Capture2
During C&D, the refueling speed buttons - Instant, Fast and Real - are all enabled.

Capture1
On the Runway, before TO, only the Instant button becomes available.

## Additional context

Discord username (if different from GitHub): TazX [Z+2]#0405

## Testing instructions

A. Start the aircraft parked C&D
   i. Turn ON the FlyPad and navigate to the Dispatch -> Fuel page
  ii. Check that all three Refuel Time buttons are available (all with white text and if pressed, they become active)
 iii. Test the aircraft refuels/ defuels in all three options (moving the fuel slider and pressing the "Set" button)

B. Start on the Runway (or any other flight phase)
   i. Turn ON the FlyPad and navigate to the Dispatch -> Fuel page
  ii. Check that ONLY the Instant Refuel Time button is available; the text in Fast and Real should be degraded (inactive)
 iii. Test the aircraft refuels/ defuels with the Instant option
 iv. When pressing Fast and Real options nothing should happen and the aircraft should not refuel 

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Comment by Benjozork** [comment]
Please fix lint

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -6,8 +6,10 @@
 
 ## 0.8.0
 1. [SOUND] Pack sounds use new APU model variables - @tracernz (Mike)
+1. [EFB] Added in-flight refueling capabilities - @Taz5150 (TazX [Z+2]#0405)
 1. [EFB] Moved setting for MCDU keyboard input into the EFB - @2hwk (2Cas#1022)
 
+
```
Comment (Benjozork) [inline .github/CHANGELOG.md:12]:
```suggestion
```

File: src/instruments/src/EFB/Dispatch/Pages/FuelPage.tsx [hunk 2]
```diff
@@ -209,6 +211,25 @@ export const FuelPage = () => {
         }
     };
 
+    const refuelButtonStatus = () => {
+        if (simGroundSpeed > 0.1 || eng1Running || eng2Running || !isOnGround || (!busDC2 && !busDCHot1)) {
```
Comment (Benjozork) [inline src/instruments/src/EFB/Dispatch/Pages/FuelPage.tsx:215]: This condition is already present above. We should store it in a variable instead.

Comment (Taz5150) [reply src/instruments/src/EFB/Dispatch/Pages/FuelPage.tsx]: Done.

**Review by Benjozork — APPROVED** [review]

**Comment by TheSquirtingPantin** [comment]
Quality Assurance Tester

Discord: Remi#8871
Object of testing: #5775
Tier of Testing: 1
Date: 8/09/2021

Testing Process:
1) Full flight performed (KIAD-KMIA).
2) Observed logic of refueling EFB page pre and post engine start
3) Instant refuel in-flight.
3) Refuelling performed post flight.

Negatives: None

Conclusions: PASS. No issues detected and EFB Refuel page is using proper logic for Real, Fast and Instant refuel. Recommending the INSTANT to be locked with a "green background" when engines are runnings and REAL and FAST options are greyed out.

**Comment by Taz5150** [comment]
> Quality Assurance Tester
> 
> Discord: Remi#8871
> Object of testing: #5775
> Tier of Testing: 1
> Date: 8/09/2021
> 
> Testing Process:
> 
> 1. Full flight performed (KIAD-KMIA).
> 2. Observed logic of refueling EFB page pre and post engine start
> 3. Instant refuel in-flight.
> 4. Refuelling performed post flight.
> 
> Negatives: None
> 
> Conclusions: PASS. No issues detected and EFB Refuel page is using proper logic for Real, Fast and Instant refuel. Recommending the INSTANT to be locked with a "green background" when engines are runnings and REAL and FAST options are greyed out.

Thank you for the test. Your suggestions makes sense. Before final approval I will make the change to lock the INSTANT background when engines running.

**Comment by NUTTTTTTTT** [comment]
Quality Assurance Tester Report

Discord: █▀█ █▄█ ▀█▀#2123
Object of testing: #PR 5775
Tier of Testing: 1
Date: 11/09/2021

Testing Process:
Loaded in on the runway at EGSS:
- Checked flyPad refueling screen, INSTANT is the only avail option, other options not clickable and greyed out, show unnavailable  if already selected until changing to instant when "ready to start" shows instead. - PASS
- Tested refuel/defueling with all three time settings, all worked as intended - PASS
- Checked fuel levels and graphic correctly displayed fuel - PASS
- Took off, retried fuel changes in multiple stages of flight, note: it is possible to remove all fuel and initiate an instant dual engine loss now. - PASS

Reloaded at the gate C+D:
- Checked flyPad refueling screen, all three options available. Tested each option again and working as intended. - PASS
- While refueling, started pushback, refuel stopped - PASS 
- Once pushed back, INSTANT is again the only avail option, others greyed out as before. With real time selected before starting pushback, refuel showed "Unavailable" until selecting INSTANT again. - PASS

Negatives: N/A
Testing Results: PASS 

Conclusions: Worked exactly as expected following testing instructions, no other issues observed, checked over functions and all seem to be okay. Great stuff! Adds a little extra flexability while keeping the realism priority. 


Media:
image

**Pull request merged**